crates/extensions/c8y_mapper_ext/src/converter.rs

```diff
@@ -2288,6 +2296,33 @@ pub(crate) mod tests {
         assert!(converter.convert(&big_event_message).await.is_empty());
     }
 
+    #[tokio::test]
+    async fn test_convert_big_event_for_child_device() {
+        let tmp_dir = TempTedgeDir::new();
+        let (mut converter, mut http_proxy) = create_c8y_converter(&tmp_dir).await;
+        tokio::spawn(async move {
+            if let Some(C8YRestRequest::CreateEvent(_)) = http_proxy.recv().await {
+                http_proxy
+                    .send(Ok(c8y_http_proxy::messages::C8YRestResponse::EventId(
+                        "event-id".into(),
+                    )))
+                    .await
+                    .unwrap()
+            }
+        });
+
+        let event_topic = "te/device/child1///e/click_event";
+        let big_event_text = create_packet((16 + 1) * 1024); // Event payload > size_threshold
+        let big_event_payload = json!({ "text": big_event_text }).to_string();
+        let big_event_message = Message::new(&Topic::new_unchecked(event_topic), big_event_payload);
+
+        let mqtt_messages = converter.convert(&big_event_message).await;
```

Comment:
That looks weird. When the event is big, we shouldn't be getting any MQTT messages, right? It should have been an HTTP message. What am I missing here?

Reply:
Here child device related messages will be received. That's why it checks the outgoing MQTT topic in assert.